helperFunctions/subscribeHelpers/SalesforceAgentAssignedHelpers/SubscribeToLiveAgentHelper.ts

```diff
@@ -69,24 +76,37 @@ export class SubscribeToLiveAgent {
 						if (chatEndedReason === 'agent') {
 							await updateRoomCustomFields(this.data.room.id, { agentEndedChat: true }, this.read, this.modify);
 							this.modify.getAnalytics().sendEvent(getEventData(this.data.room.id, EventName.CHAT_CLOSED_BY_AGENT));
+							await handleEndChatCallback.handleEndChat(CloseMessages.CLOSED_BY_AGENT);
+						} else if (chatEndedReason !== 'client') {
+							const event = getEventData(this.data.room.id, EventName.CHAT_CLOSED_BY_UNDEFINED_REASON);
```

Comment:
Why do have an `undefined` reason here? Can you check for possible reasons and add those instead?

Reply:
Do we have an Enum of reasons? I'm trying to find where we set this parameter.

Reply:
It comes from Salesforce API